text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+Component authors can use the `connectedCallback` of the dynamic component to signal when the component has been rendered to the DOM.
+
+#### Assigning attributes and event listeners
+
+Attributes and event listeners can be passed to the `<lwc.dynamic>` tag declaratively or can be assigned directly to the custom element imperatively.
+
+Attributes and event listeners assigned declaratively will be assigned to the dynamic component once it has been created.
+
+Declaratively:
+
+```html
+<template>
+    <lwc.dynamic lwc:is={ctor} class="container" onclick={handleClick}></lwc.dynamic>
+</template>
+```
+
+Imperatively:
+
+```javascript
+import { LightningElement } from "lwc";
+
+export default class DynamicCtor extends LightningElement {
+  customCtor;
+
+  connectedCallback() {
+    this.loadCtor();
+  }
+
+  async loadCtor() {
+    const { default: ctor } = await import("c/customConstructor");
+    this.customCtor = ctor;
+    this.template.querySelector(ctor.lightningElementName).setAttribute('title', 'hawaii');
+    this.template.querySelector(ctor.lightningElementName).addEventListener('click', () => console.log('hello world'));
+  }
+}
+```
+#### Usage with other LWC directives
+
+All [LWC custom element directives](https://lwc.dev/guide/reference#directives-for-nested-templates) with the exception of `lwc:external` are available to use on `<lwc.dynamic>` and their functionality will be passed through to the custom element once it has been created.
+
+`lwc:external` will not be available because the constructor provided to the `lwc:is` directive must be a `LightningElementConstructor`.
+
+#### Children of the dynamically created element
+
+The `<lwc.dynamic>` tag accepts child elements, first creating the dynamic component and subsequently the children afterwards.  Each time the dynamic component changes, the existing dynamic component will be unmounted along with all of its children.  The new dynamic component will be created along with the children.
+
+```html
+<template>
+    <lwc.dynamic lwc:is={ctor}>
+        <span>child</span>
+    </lwc.dynamic>
+</template>
+```
+
+#### Light DOM vs shadow DOM
+
+A dynamic component can switch between light DOM and shadow DOM and should follow the same semantics as currently exists with other forms of dynamic rendering components.
+
+For example
+
+```html
+<template>
+    <template lwc:if={renderLight}>
+        <light-component></light-component>
+    <template lwc:else>
+        <shadow-component></shadow-component>
+    </template>
+</template>
+```
+
+Is semantically the same as
+
+```html
+<template>
+    <!-- The ctor swaps between light and shadow DOM components -->
+    <lwc.dynamic lwc:is={ctor}></lwc.dynamic>
+</template>
+```
+
+In both cases the dynamic component will be fully mounted and unmounted.
```

Comment:
This same challenge applies to synthetic vs native shadow components, since they have different slot mechanisms.